### ISSUE TITLE
fp16 channels_last created Nan in batchnorm backward

### ISSUE THREAD

**Issue by jthakurH** [pytorch/pytorch#147303]
### 🐛 Describe the bug

If input is set with channels_last then there is Nan in output of backward

```
import torch


op = torch.nn.BatchNorm2d(num_features=3, affine=True, track_running_stats=True)
ifm = torch.empty(size=[16, 3, 224, 224]).uniform_(0, 1).to(dtype=torch.float16)
ifm = ifm.contiguous(memory_format=torch.channels_last) # this creates Nan in output
ifm = ifm.requires_grad_()

res = op(ifm)
bwd_tensor = torch.empty(size=res.shape).uniform_(0, 1).to(dtype=torch.float16)

res.backward(bwd_tensor)

print(ifm.grad)

```

Output
```
tensor([[[[nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          ...,
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan]],

         [[nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
          [nan, nan, nan,  ..., nan, nan, nan],
```

### Versions

[pip3] numpy==1.23.5
[pip3] torch==2.6.0

cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10
